### PR TITLE
Adding TSLint and fixing issues.

### PR DESCRIPTION
**Pull request by pschutz93**
This pull request adds TSLint to `glimmer-application`. I pulled the `tslint.json` from [glimmer-vm](https://github.com/glimmerjs/glimmer-vm) and fixed existing style issues.

### CODE REVIEW

File: testem.json [hunk 1]
```diff
@@ -3,5 +3,5 @@
   "test_page": "index.html",
   "src_files": ["src/**/*", "test/**/*"],
   "disable_watching": true,
-  "launch_in_ci": ["Firefox", "Chrome"]
```
Comment (twokul) [inline testem.json]: you might want to bring `Firefox` back

**Comment by pschutz93** [comment]
Added `Firefox` back @twokul

File: src/helpers/action.ts [hunk 2]
```diff
@@ -34,7 +34,7 @@ export function debugInfoForReference(reference: any): string {
   let parent;
   let property;
 
-  if (reference == null) { return message; }
```
Comment (rwjblue) [inline src/helpers/action.ts]: Hmm.  I think this `==` may have been intentional (as it matches `null` and `undefined` but not other falsey values).  I think this should be updated to `reference === null || reference === undefined`.

File: src/helpers/action.ts [hunk 3]
```diff
@@ -53,7 +53,7 @@ export function debugInfoForReference(reference: any): string {
 
 function debugName(obj: any) {
   let objType = typeof obj;
-  if (obj == null) {
```
Comment (rwjblue) [inline src/helpers/action.ts]: Same as above, I think this should be updated to `reference === null || reference === undefined`.

**Comment by rwjblue** [comment]
Thank you!

**Pull request merged**